src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -115,6 +118,20 @@ public void OpenPin(int pinNumber, PinMode mode)
             SetPinMode(pinNumber, mode);
         }
 
+        /// <summary>
+        /// Opens a pin and sets it to a specific mode and value.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
+        /// <param name="mode">The mode to be set.</param>
+        /// <param name="initialValue">The initial value to be set if the mode is output. The driver will attempt to set the mode without causing glitches to the other value.
+        /// (if <paramref name="initialValue"/> is <see cref="PinValue.High"/>, the pin should not glitch to low during open)</param>
+        public void OpenPin(int pinNumber, PinMode mode, PinValue initialValue)
+        {
+            OpenPin(pinNumber);
+            Write(pinNumber, initialValue);
```

Comment:
Do you know what is the initial pin mode when opening one for a driver? If it is output, then this method would actually light the LED even when called with mode=input right? If that's the case, perhpas we should call setPinMode first?

Reply:
That's a good point. The expectation is that opening leaves the mode as-is (but that seems to be inconsistent between drivers, see also #1479), but then yes, if it was output, and the new mode is input, it would change the value first.